src/ynEIGEN/withdrawalsProcessor.sol

```diff
@@ -0,0 +1,251 @@
+        delegationManager = IDelegationManager(_delegationManager);
+
+        minPendingWithdrawalRequestAmount = 0.1 ether;
+
+        // @todo - fix that
+        IERC20[] memory _assets = assetRegistry.getAssets();
+        for (uint256 i = 0; i < _assets.length; ++i) {
+            minShares[_assets[i]] = 1 ether;
+        }
+    }
+
+    //
+    // Processor functions
+    //
+
+    /// @notice Queues withdrawals
+    /// @dev Reverts if the total pending withdrawal requests are below the minimum threshold
+    /// @dev Skips nodes with shares below the minimum threshold
+    /// @dev If a nodes has more shares than the minimum threshold, exits the node entirely
+    /// @dev Tries to satisfy the pending withdrawal requests while prioritizing withdrawals in one asset
+    /// @dev Saves the queued withdrawals together in a batch, to be completed in the next step (`completeQueuedWithdrawals`)
+    /// @return True if all pending withdrawal requests were queued, false otherwise
+    function queueWithdrawals() external onlyOwner returns (bool) {
+
+        uint256 _minPendingWithdrawalRequestAmount = minPendingWithdrawalRequestAmount;
+        uint256 _pendingWithdrawalRequests = withdrawalQueueManager.pendingRequestedRedemptionAmount() - totalQueuedWithdrawals;
+        if (_pendingWithdrawalRequests <= minPendingWithdrawalRequestAmount) revert PendingWithdrawalRequestsTooLow();
+
+        totalQueuedWithdrawals += _pendingWithdrawalRequests;
+
+        ITokenStakingNode[] memory _nodes = tokenStakingNodesManager.getAllNodes();
+        IERC20[] memory _assets = assetRegistry.getAssets();
+
+        uint256 _queuedIdBefore = queuedId;
+        uint256 _assetsLength = _assets.length;
+        uint256 _nodesLength = _nodes.length;
+        for (uint256 i = 0; i < _assetsLength; ++i) {
+            uint256 _minShares = minShares[_assets[i]];
+            IStrategy _strategy = ynStrategyManager.strategies(_assets[i]);
+            for (uint256 j = 0; j < _nodesLength; ++j) {
+                address _node = address(_nodes[j]);
+                uint256 _nodeShares = _strategy.shares(_node);
+                if (_nodeShares > _minShares) {
+                    uint256 _unitWithdrawalAmount = _sharesToUnit(_nodeShares, _assets[i], _strategy);
+                    _pendingWithdrawalRequests =
+                        _unitWithdrawalAmount < _pendingWithdrawalRequests ? _pendingWithdrawalRequests - _unitWithdrawalAmount : 0;
+                    queuedWithdrawals[queuedId++] = QueuedWithdrawal(
+                        _node,
+                        address(_strategy),
+                        delegationManager.cumulativeWithdrawalsQueued(_node), // nonce
+                        _nodeShares,
+                        uint32(block.number), // startBlock
+                        false // completed
+                    );
+                    ITokenStakingNode(_node).queueWithdrawals(_strategy, _nodeShares);
```

Comment:
in retrospect we should have made this function here take arrays since the eigenlayer interface takes arrays:

https://github.com/yieldnest/yieldnest-protocol/blob/main/src/ynEIGEN/TokenStakingNode.sol#L141

Would save a lot of gas, because there's many calls stacked here through multiple contracts

Reply:
We can do that as another upgrade